src/core/src/test/java/org/apache/jmeter/gui/util/JSyntaxTextAreaTest.java

```diff
@@ -41,7 +45,37 @@ public void testSetLanguage() {
             textArea.setLanguage(null);
             assertEquals(SyntaxConstants.SYNTAX_STYLE_NONE, textArea.getSyntaxEditingStyle());
         } catch (HeadlessException he) {
-            // Does not work in headless mode
+            // Does not work in headless mode, which depends on value of java.awt.headless property
+            // and the OS (e.g. might work on MacOS and not on Linux due to missing X11).
+            System.out.println("WARNING for JSyntaxTextAreaTest.testSetLanguage test: does not work in headless mode");
+        }
+    }
+
+    @Test
+    public void testHeadless() {
+        String key = "java.awt.headless";
+        String initialValue = System.getProperty(key);
+        try {
+            System.setProperty(key, "true");
```

Comment:
Is this really a sane way to test headless mode?

Reply:
Maybe there is a better way? It is indeed tricky as it might impact other tests, that's why the `@Isolated` is added. The code directly depends on the property value in its behaviour. I've seen some mock framework code that overrides the System.getProperty() to return a certain value, but there might be classloading issues.

Reply:
For the other tests, we are expecting the test suite to be set up in the correct way (for example most of our CI stuff is done in a headless setup). But if this works, we can use it.